core/services/scan_rate_limit_test.go

```
@@ -0,0 +1,61 @@
+	// We assert that their cache keys SHOULD be different so they don't block each other.
+	require.NotEqual(t, rateLimitCacheKey(unauthenticatedWorkload), rateLimitCacheKey(authenticatedWorkload),
+		"Expected cache keys to differ when credentials are provided, preventing cross-tenant DoS")
+
+	// 3. To demonstrate the DoS effect on ScanRegistry:
+	platform := &recordingPlatform{}
+	countingSBOM := &countingSBOMCreator{SBOMCreator: adapters.NewMockSBOMAdapter(false, false, false)}
+	cveAdapter := adapters.NewMockCVEAdapter()
+	s := NewScanService(countingSBOM, repositories.NewMemoryStorage(false, false), cveAdapter, repositories.NewMemoryStorage(false, false), platform, nil, false, false, true, false, false)
+
+	ctx := context.TODO()
+	ctx = enrichContext(ctx, unauthenticatedWorkload, s.Version())
```

golangci-lint (2.12.2)

[error] ineffectual assignment to ctx

(ineffassign)